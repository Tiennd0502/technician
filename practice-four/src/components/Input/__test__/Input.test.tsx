import { render } from '@testing-library/react';
import Input from '..';

describe('Input component', () => {
  test('should show error message when reive error', () => {
    const props = {
      error: 'Error message',
    };

    const { getByText } = render(<Input {...props} />);
    const error = getByText('Error message');

    expect(error).toBeTruthy();
  });

  test('should show label', () => {
    const props = {
      label: 'Name',
    };
    const { getByText } = render(<Input {...props} />);
    const target = getByText('Name');

    expect(target).toBeTruthy();
  });

  test('match snapshot', () => {
    const props = {
      error: 'Error message',
      label: 'Name',
    };
    const { container } = render(<Input {...props} />);

    expect(container).toMatchSnapshot();
  });
});
