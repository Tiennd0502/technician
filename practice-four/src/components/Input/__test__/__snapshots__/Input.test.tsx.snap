// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component match snapshot 1`] = `
<div>
  <div
    class="chakra-form-control css-0"
    data-invalid=""
    role="group"
  >
    <label
      class="chakra-form__label css-gu124"
      data-invalid=""
      for="field-:r2:"
      id="field-:r2:-label"
    >
      Name
    </label>
    <input
      aria-describedby="field-:r2:-feedback"
      aria-invalid="true"
      class="chakra-input css-jgxprw"
      id="field-:r2:"
      type="text"
    />
    <div
      aria-live="polite"
      class="chakra-form__error-message css-70qvj9"
      id="field-:r2:-feedback"
    >
      Error message
    </div>
  </div>
</div>
`;
